digital_asset_types/src/dao/scopes/asset.rs

```diff
@@ -289,3 +289,53 @@ pub async fn get_by_id(
         groups: grouping,
     })
 }
+
+pub async fn fetch_transactions(
+    conn: &impl ConnectionTrait,
+    tree: Vec<u8>,
+    leaf_id: i64,
+    pagination: &Pagination,
+    limit: u64,
+) -> Result<Vec<Vec<String>>, DbErr> {
+    let mut stmt = cl_audits::Entity::find()
+        .filter(cl_audits::Column::Tree.eq(tree))
+        .filter(cl_audits::Column::LeafIdx.eq(leaf_id))
+        .order_by(cl_audits::Column::CreatedAt, sea_orm::Order::Desc);
+
+    stmt = paginate(pagination, limit, stmt);
+    let transactions = stmt.all(conn).await?;
+    let transaction_list: Vec<Vec<String>> = transactions
+        .into_iter()
+        .map(|transaction| vec![transaction.tx, transaction.instruction])
+        .collect();
+
+    Ok(transaction_list)
+}
+
+pub async fn get_transactions(
+    conn: &impl ConnectionTrait,
+    asset_id: Vec<u8>,
+    pagination: &Pagination,
+    limit: u64,
+) -> Result<Vec<Vec<String>>, DbErr> {
+    let mut stmt = asset::Entity::find().distinct_on([(asset::Entity, asset::Column::Id)]);
```

Comment:
Why not do a join instead?

Reply:
Because of performance, `cl_audits` is huge and when I did a describe, doing it without join was better.